packages/elasticsearch-plugin/src/indexing/indexer.controller.ts

```diff
@@ -335,42 +335,97 @@ export class ElasticsearchIndexerController implements OnModuleInit, OnModuleDes
                     }
                 }
 
-                const deletedProductIds = await this.connection
+                const totalDeletedProductIds = await this.connection
                     .getRepository(Product)
                     .createQueryBuilder('product')
-                    .select('product.id')
                     .where('product.deletedAt IS NOT NULL')
-                    .getMany();
+                    .getCount();
 
-                for (const { id: deletedProductId } of deletedProductIds) {
-                    operations.push(...(await this.deleteProductOperations(ctx, deletedProductId)));
-                }
-
-                const productIds = await this.connection
+                const totalProductIds = await this.connection
                     .getRepository(Product)
                     .createQueryBuilder('product')
-                    .select('product.id')
                     .where('product.deletedAt IS NULL')
-                    .getMany();
+                    .getCount();
+
+                // Run delete operation by bulk
+                Logger.verbose(`Will reindex ${totalDeletedProductIds} deleted products`, loggerCtx);
+
+                let deletedProductIds = [];
+                let skip = 0;
+                let finishedDeletedItem = 0;
+                do {
+                    const operations: BulkVariantOperation[] = [];
+                    deletedProductIds = await this.connection
+                        .getRepository(Product)
+                        .createQueryBuilder('product')
```

Comment:
`skip` & `take` are not being used in this query, so this part is not currently being batched.

Reply:
My bad I have it on my own code, I missed it when creating the PR. I add it